dirty_equals/_other.py

```diff
@@ -145,3 +147,85 @@ def is_even(x):
 
     def equals(self, other: Any) -> bool:
         return self.func(other)
+
+
+class IsUrl(DirtyEquals[AnyUrl]):
+    """
+    A class that checks if a value is a valid URL, optionally checking different URL types and attributes with
+    [Pydantic](https://pydantic-docs.helpmanual.io/usage/types/#urls).
+    """
+
+    allowed_attribute_checks: Set[str] = {
+        'scheme',
+        'host',
+        'host_type',
+        'user',
+        'password',
+        'tld',
+        'port',
+        'path',
+        'query',
+        'fragment',
+    }
+
+    def __init__(
+        self,
+        url_type: Literal[
+            'AnyUrl', 'AnyHttpUrl', 'HttpUrl', 'FileUrl', 'PostgresDsn', 'AmpqpDsn', 'RedisDsn'
+        ] = 'AnyUrl',
+        **expected_attributes: Any,
```

Comment:
some duplication, but much easier for users if you list each kwarg individually here - so IDEs can provide auto-complete.